postcard-bindgen-core/src/code_gen/js/mod.rs

```diff
@@ -116,14 +132,15 @@ impl Default for GenerationSettings {
             runtime_type_checks: false,
             type_script_types: false,
             module_structure: true,
+            esm_module: false,
         }
     }
 }
 
 pub fn generate(
     mut containers: ContainerCollection,
     gen_settings: impl Borrow<GenerationSettings>,
-) -> Exports<JavaScript> {
+) -> (Exports<JavaScript>, bool) {
```

Comment:
Instead of only returning a bool, you can add a struct wrapping it.

```rust
struct ExportMeta {
   pub esm_module: bool
}
```

In rust, there is the idea of wrapping types into structs to make the type more verbose. In general, returning a tuple is fine when making the type verbose as said.